### PR TITLE
Fix issue 18097 - unittest symbol names can be used before semantic

### PR DESCRIPTION
**Pull request by atilaneves**
Not only does this PR fix issue 18097, it also simplifies the unittest identifier logic and fixes the test for related  issue 16995.

### CODE REVIEW

**Comment by JinShil** [comment]
Let me know when this is no longer a WIP

**Comment by atilaneves** [comment]
@JinShil No longer WIP and ready for review.

File: test/runnable/issue16995.d [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// COMPILE_SEPARATELY
+// EXTRA_SOURCES: imports/module_with_tests.d
+
+import imports.module_with_tests;
+import core.exception: AssertError;
+
+shared static this()
+{
+    import core.runtime: Runtime;
+    Runtime.moduleUnitTester = () => true;
```
Comment (wilzbach) [inline test/runnable/issue16995.d]: Returning `bool` is deprecated, return a number instead.
See also: https://github.com/dlang/druntime/pull/1685

**Comment by atilaneves** [comment]
@wilzbach Return `UnitTestResult` then, not an int. Changed.

**Review by JinShil — APPROVED** [review]

**Review by ibuclaw — APPROVED** [review]

File: src/dmd/func.d [hunk 2]
```diff
@@ -3476,56 +3473,17 @@ extern (C++) final class UnitTestDeclaration : FuncDeclaration
         return FuncDeclaration.syntaxCopy(utd);
     }
 
-    /**
-       Sets the "real" identifier, replacing the one created in the contructor.
-       The reason for this is that the "real" identifier can only be generated
-       properly in the semantic pass. See:
-       https://issues.dlang.org/show_bug.cgi?id=16995
-     */
-    final void setIdentifier()
```
Comment (ibuclaw) [inline src/dmd/func.d]: Was this added to headers also?

Comment (atilaneves) [reply src/dmd/func.d]: I grepped and didn't find it anywhere.

**Pull request merged**